packages/api-client/__tests__/api/updateAddressDetails.spec.ts

```diff
@@ -0,0 +1,77 @@
+import updateAddressDetails from '../../src/api/updateAddressDetails';
+import setOrderBillingAddressMutation from './../../src/api/updateAddressDetails/setOrderBillingAddressMutation';
+import setOrderShippingAddressMutation from './../../src/api/updateAddressDetails/setOrderShippingAddressMutation';
+import { Context } from '../../src/types';
+import { BILLING_TYPE } from '../../src/helpers/constants';
+
+describe('[vendure-api-client] updateAddressDetails', () => {
+  it('updates shipping details', async () => {
+    const givenVariables = {
+      input: {
+        fullName: 'Eliot Anderson',
+        company: 'F Society',
+        streetLine1: '1st Avenue',
+        streetLine2: '1st Avenue',
+        city: 'New York',
+        province: 'New York',
+        postalCode: '34-234',
+        countryCode: 'US',
+        phoneNumber: '123123123',
+        defaultShippingAddress: true,
+        defaultBillingAddress: false
+      }
+    };
+
+    const context = ({
+      config: {},
+      client: {
+        mutate: ({ variables, mutation }) => {
+          expect(variables).toEqual(givenVariables);
+          expect(mutation).toEqual(setOrderShippingAddressMutation);
+
+          return { data: 'update shipping details response' };
+        }
+      },
+      extendQuery: (customQuery, args) => args
+    } as unknown) as Context;
+
+    const { data } = await updateAddressDetails(context, { input: givenVariables.input });
+
+    expect(data).toBe('update shipping details response');
```

Comment:
you can extract string into a variable and place it into a given section with the name `expectedAddressDetails` this will contribute to readability.

Reply:
Good call!